libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFMutableDictionaryThreadSafe.h

```diff
@@ -0,0 +1,88 @@
+/*
+ SFMutableDictionaryThreadSafe.h
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on Wed Oct 21 17:47:00 PDT 2016.
+ 
+ Copyright (c) 2016-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ This class provides a thread safe Mutable Dictionary. Allows for concurrent reads and exclusive writes.
+ Employs the use of dispatch_sync for reads and dispatch_barrier_async for writes to an
+ internally managed NSMutableDictionary. dispatch_barrier_async() introduces a synchronization
+ point allowing previously enqueued blocks to complete. All blocks placed after the barrier will
+ execute oinly after the barrier block is completed.
+ */
+#import <Foundation/Foundation.h>
+
+@interface SFMutableDictionaryThreadSafe : NSObject
```

Comment:
I agree that the wrapper may not be required. The SFSmartStorePlugin.m seems to have   mutable state. IMO it might be better to wrap it into a dispatch_sync. Will refactor code to move to serial-Q locally.